superset-frontend/src/SqlLab/components/TemplateParamsEditor.jsx

```diff
@@ -68,15 +60,19 @@ export default class TemplateParamsEditor extends React.Component {
     } catch (e) {
       isValid = false;
     }
-    this.setState({ parsedJSON, isValid, codeText });
+    setState({ parsedJSON, isValid, codeText });
     const newValue = isValid ? codeText : '{}';
-    if (newValue !== this.props.code) {
-      this.props.onChange(newValue);
+    if (newValue !== props.code) {
+      props.onChange(newValue);
     }
-  }
+  };
+
+  useEffect(() => {
+    onChange(codeText);
+  }, [codeText]);
 
-  renderDoc() {
-    return (
+  const renderDoc = () => {
```

Comment:
this can be pulled out the of functional component since it doesn't rely on any state